lib/api/api-request.js

```diff
@@ -5,15 +5,14 @@ const { AsyncResource } = require('node:async_hooks')
 const { Readable } = require('./readable')
 const { InvalidArgumentError, RequestAbortedError } = require('../core/errors')
 const util = require('../core/util')
-const { getResolveErrorBodyCallback } = require('./util')
 
 class RequestHandler extends AsyncResource {
   constructor (opts, callback) {
     if (!opts || typeof opts !== 'object') {
       throw new InvalidArgumentError('invalid opts')
     }
 
-    const { signal, method, opaque, body, onInfo, responseHeaders, throwOnError, highWaterMark } = opts
+    const { signal, method, opaque, body, onInfo, responseHeaders, highWaterMark } = opts
 
```

Comment:
throw error if `throwError === true`

Reply:
Shall we? Its a major, it means overall that it won't work if not documented.

Reply:
Doesn't hurt? Less suprising to people that don't update their code.

Reply:
Fair: [f723cc2](https://github.com/nodejs/undici/pull/3451/commits/f723cc25269b4134dc774d14c52f9a1e7663150d)